From 029435236c08e79f66f1461f8e684ea690c1b7cd Mon Sep 17 00:00:00 2001
From: Harald Wellmann <hwellmann@apache.org>
Date: Tue, 16 Jun 2015 10:52:03 +0200
Subject: [PATCH] DELTASPIKE-931 Integration tests fail when dependencies are
 class folders, not JARs

make library names unique by appending a numeric suffix, if needed
---
 .../deltaspike/test/utils/ShrinkWrapArchiveUtil.java      | 8 ++++++--
 1 file changed, 6 insertions(+), 2 deletions(-)

diff --git a/deltaspike/test-utils/src/main/java/org/apache/deltaspike/test/utils/ShrinkWrapArchiveUtil.java b/deltaspike/test-utils/src/main/java/org/apache/deltaspike/test/utils/ShrinkWrapArchiveUtil.java
index bbdd06834..d2d404387 100644
--- a/deltaspike/test-utils/src/main/java/org/apache/deltaspike/test/utils/ShrinkWrapArchiveUtil.java
+++ b/deltaspike/test-utils/src/main/java/org/apache/deltaspike/test/utils/ShrinkWrapArchiveUtil.java
@@ -75,19 +75,23 @@ public static JavaArchive[] getArchives(ClassLoader classLoader,
             Enumeration<URL> foundFiles = classLoader.getResources(markerFile);
 
             List<JavaArchive> archives = new ArrayList<JavaArchive>();
+            int numArchives = 0;
 
             while (foundFiles.hasMoreElements())
             {
                 URL foundFile = foundFiles.nextElement();
                 LOG.fine("Evaluating Java ClassPath URL " + foundFile.toExternalForm());
+                String suffix = (numArchives == 0) ? "" : Integer.toString(numArchives);
 
                 JavaArchive archive
-                    = createArchive(foundFile, markerFile, includeIfPackageExists, excludeIfPackageExists, archiveName);
+                    = createArchive(foundFile, markerFile, includeIfPackageExists, excludeIfPackageExists, 
+                        archiveName + suffix);
                 if (archive != null)
                 {
                     LOG.info("Test " + getTestName()
                             + " Adding Java ClassPath URL as JavaArchive " + foundFile.toExternalForm());
                     archives.add(archive);
+                    numArchives++;
                 }
             }
 
@@ -236,7 +240,7 @@ private static List<File> collectArchiveEntries(File archiveBasePath)
             return archiveEntries;
         }
 
-        return Collections.EMPTY_LIST;
+        return Collections.emptyList();
     }
 
 
